### PR TITLE
Validation of Id on ExtRef relation.

### PR DESCRIPTION
**Pull request by jacobchristensen79**
Configurable option to enable the validation on extref $ref references to existing objects.

### CODE REVIEW

**Comment by hairmare** [comment]
AFAIK there are some cases where we only want the format of the extref to be valid even though it may point to a 404. It looks like that won't work anymore if we validate all the things.

**Comment by hairmare** [comment]
I'm still not sure that we should not allow records pointing to a 404. Also, this certainly needs some tests to cover the functionality as well as check the error message.

File: src/Graviton/RestBundle/Validator/Constraints/ExtReference/ExtReferenceValidator.php [hunk 1]
```diff
@@ -46,6 +65,19 @@ public function validate($value, Constraint $constraint)
             !in_array($value->getRef(), $constraint->collections, true)
         ) {
             $this->context->addViolation($constraint->message, ['%collection%' => $value->getRef()]);
+            return;
         }
+
+        // Does ID exists in DB.
+        if ($this->validateId && !in_array($value->getRef(),['App', 'TestCaseReadOnly'])) {
```
Comment (narcoticfresh) [inline src/Graviton/RestBundle/Validator/Constraints/ExtReference/ExtReferenceValidator.php]: wtf? `TestCaseReadOnly`

**Comment by narcoticfresh** [comment]
@jacobchristensen79 sorry for joining the discussion so late ;-)

I can't see if this still enables to save `extref`s that lead to non-existing documents ( @hairmare 's question) . Cause that feature is crucial - just imagine the mess in initialdata import ;-) 

So what does it actually change? Do you intend to set the config flag to `false` anyway all the time and just have it around for debug scenarios?

**Pull request closed without merging**